server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala

```diff
@@ -0,0 +1,135 @@
+    ) (model.Invocation.apply _)
+
+  private implicit val invocationWrite: Writes[Invocation] = (invocation: Invocation) =>
+    Json.arr(invocation.methodName, invocation.arguments, invocation.methodCallId)
+
+  // RequestObject
+  private implicit val capabilityIdentifierWrites: Format[CapabilityIdentifier] = Json.valueFormat[CapabilityIdentifier]
+  private implicit val requestObjectRead: Format[RequestObject] = Json.format[RequestObject]
+
+  // ResponseObject
+  private implicit val responseObjectFormat: Format[ResponseObject] = Json.format[ResponseObject]
+
+  private implicit val maxSizeUploadWrites: Writes[MaxSizeUpload] = Json.valueWrites[MaxSizeUpload]
+  private implicit val maxConcurrentUploadWrites: Writes[MaxConcurrentUpload] = Json.valueWrites[MaxConcurrentUpload]
+  private implicit val maxSizeRequestWrites: Writes[MaxSizeRequest] = Json.valueWrites[MaxSizeRequest]
+  private implicit val maxConcurrentRequestsWrites: Writes[MaxConcurrentRequests] = Json.valueWrites[MaxConcurrentRequests]
+  private implicit val maxCallsInRequestWrites: Writes[MaxCallsInRequest] = Json.valueWrites[MaxCallsInRequest]
+  private implicit val maxObjectsInGetWrites: Writes[MaxObjectsInGet] = Json.valueWrites[MaxObjectsInGet]
+  private implicit val maxObjectsInSetWrites: Writes[MaxObjectsInSet] = Json.valueWrites[MaxObjectsInSet]
+
+  private implicit val maxMailboxesPerEmailWrites: Writes[MaxMailboxesPerEmail] = Json.valueWrites[MaxMailboxesPerEmail]
+  private implicit val maxMailboxDepthWrites: Writes[MaxMailboxDepth] = Json.valueWrites[MaxMailboxDepth]
+  private implicit val maxSizeMailboxNameWrites: Writes[MaxSizeMailboxName] = Json.valueWrites[MaxSizeMailboxName]
+  private implicit val maxSizeAttachmentsPerEmailWrites: Writes[MaxSizeAttachmentsPerEmail] = Json.valueWrites[MaxSizeAttachmentsPerEmail]
+  private implicit val mayCreateTopLevelMailboxWrites: Writes[MayCreateTopLevelMailbox] = Json.valueWrites[MayCreateTopLevelMailbox]
+
+  private implicit val usernameWrites: Writes[Username] = username => JsString(username.asString)
+  private implicit val urlWrites: Writes[URL] = url => JsString(url.toString)
+  private implicit val coreCapabilityWrites: Writes[CoreCapabilityProperties] = Json.writes[CoreCapabilityProperties]
+  private implicit val mailCapabilityWrites: Writes[MailCapabilityProperties] = Json.writes[MailCapabilityProperties]
+
+  private implicit def setCapabilityWrites(implicit corePropertiesWriter: Writes[CoreCapabilityProperties],
+                                   mailCapabilityWrites: Writes[MailCapabilityProperties]): Writes[Set[_ <: Capability]] =
+    (set: Set[_ <: Capability]) => {
+      JsObject(set.map {
+        case capability: CoreCapability => (
+          capability.identifier.asString, corePropertiesWriter.writes(capability.properties))
+        case capability: MailCapability => (
+          capability.identifier.asString, mailCapabilityWrites.writes(capability.properties))
```

Comment:
missing wildcard check https://github.com/linagora/james-project/pull/3202#discussion_r396421853

Reply:
https://github.com/linagora/james-project/pull/3238/commits/28bf3a02836510a90a32b074eaded88584c7b7b8

Decided to ignore an unknown capability